java/Cherepnin_Oleh/src/com/zoolatech/lecture4/tasks/_1/Container.java

```diff
@@ -0,0 +1,32 @@
+package com.zoolatech.lecture4.tasks._1;
+
+public class Container {
+    private String value;
+    private ContainerUpdatesListener innerContainer;
+
+    {
+        innerContainer = new ContainerUpdatesListener();
+    }
+
+    public Container(String value) {
+        this.value = value;
+        System.out.println(innerContainer.hasValueChanged(value));
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public void setValue(String value) {
+        System.out.println(innerContainer.hasValueChanged(value));
+        this.value = value;
+    }
+
+    class ContainerUpdatesListener {
+        boolean hasValueChanged(String newValue) {
```

Comment:
the method shouldn't accept any value; find another way how you can track if an instance variable has changed